src/render/renderer.cpp

```diff
@@ -0,0 +1,132 @@
+#include "render/renderer.h"
+
+#include <iostream>
+#include <limits>
+
+namespace render
+{
+
+Renderer::Renderer( const float fov,
+                    const int imageWidth,
+                    const int imageHeight,
+                    const float nearClip,
+                    const float farClip )
+    : camera( fov, (float) imageWidth / (float) imageHeight )
+    , imageWidth( imageWidth )
+    , imageHeight( imageHeight )
+    , nearClip( nearClip )
+    , farClip( farClip )
+{
+}
+
+obj::Scene& Renderer::getScene()
+{
+    return scene;
+}
+
+Camera& Renderer::getCamera()
+{
+    return camera;
+}
+
+const obj::Scene& Renderer::getScene() const
+{
+    return scene;
+}
+
+void Renderer::setScene( const obj::Scene& newScene )
+{
+    scene = newScene;
+}
+
+void Renderer::setCamera( const Camera& newCamera )
+{
+    camera = newCamera;
+}
+
+void Renderer::render() const
+{
+    std::cout << std::endl;
+
+    for ( int y = 0; y < imageHeight; y++ )
```

Comment:
if you change imageHeight to unsigned int, change also in these loops;. also, ++y is much faster than y++, use ++y in loops

Reply:
In most cases, it doesn't matter. I know that ++y won't make a copy, and y++ might make one, but the compiler will optimize everything. Especially in this case, even if it doesn't it'll be nearly impossible to measure the real difference in performance with this low amount of iterations. So it is more of a style rather than a real deal. For me, prefix increment looks like shit, and since it has little to no impact on performance, I'd stick with postfix.